File: service/api/src/main/java/org/delivery/api/exception/model/ExceptionResponse.java
package org.delivery.api.exception.model;

import org.delivery.api.common.api.ResultType;

public record ExceptionResponse(
	Integer errorCode,
	String errorMessage
) {

	public static ExceptionResponse from(final ResultType errorType) {
		return new ExceptionResponse(errorType.getCode(), errorType.getMessage());
	}

}

File: delivery/service/common/src/main/kotlin/org/delivery/common/exception/model/ExceptionResponse.kt
package org.delivery.common.exception.model

import org.delivery.common.api.ResultType

data class ExceptionResponse(
    val errorCode: Int,
    val errorMessage: String
) {

    companion object {
        @JvmStatic
        fun from(errorType: ResultType) = ExceptionResponse(
            errorCode = errorType.code,
            errorMessage = errorType.message
        )
    }

}